### ISSUE TITLE
bug: autoWatchBatchDelay option is being "ignored"

### ISSUE THREAD

**Issue by ferrarit** [karma-runner/karma#1520]
`autoWatchBatchDelay` options is irrelevant in current version of karma

The problem seems to be caused by the use of `_.throttle`

https://github.com/karma-runner/karma/commit/fabdbaef2e91aa00ab4771cd0e307267ac6925dd#diff-e8e0621a959fd2639bcc23cd2601f67fR73

replacing `_.throttle` with `_.debounce` fixes the issue

```javascript
this._emitModified = _.debounce(function () {
    self._emitter.emit('file_list_modified', self.files)
}, this._batchInterval)
```

**Comment by Dignifiedquire** [comment]
I can't believe I did this mistake again, for some reason I'm never able to use `throttle` and `debounce` correctly. 
Thanks a lot for reporting, would you be up for a PR?

**Comment by ferrarit** [comment]
Honestly, I wouldn't know how to write the tests :)

**Comment by Dignifiedquire** [comment]
@ferrarit there are actually tests https://github.com/karma-runner/karma/blob/master/test/unit/file-list.spec.coffee#L568-L639 which start failing when I introduce `debounce`. Could you maybe create a small setup that reproduces the behaviour you are seeing?

**Comment by ferrarit** [comment]
I have created a simple test here: https://github.com/ferrarit/karma-test

1. `npm install`
2. `karma start karma.conf`
3. edit both `test/test-a.js` and `test/test-b.js` 
4. save both files at the same time
5. You should get a message like "Delaying execution, these browsers are not ready..." and tests are going to run twice

1. Replace `_.throttle` with `_.debounce`
2. edit both `test/test-a.js` and `test/test-b.js` 
3. save both files at the same time
4. tests run only once

**Comment by Dignifiedquire** [comment]
Thanks :octocat:

**Comment by ferrarit** [comment]
Thank you sir for the quick fix and the amazing job you are doing with Karma. :+1:

**Comment by Dignifiedquire** [comment]
Looks like my fix is not working yet :( Will revisit tomorrow with a fresh brain

**Comment by Merott** [comment]
Any update on this?

I have no experience in using either `throttle` or `debounce`, but I was wondering, whichever is used, shouldn't the `leading` option be set to `false`? it seems that `throttle` is more of a convenience function anyway, as it [forwards the call](https://github.com/lodash/lodash/blob/3.0.4-npm-packages/lodash.throttle/index.js#L66) to `debounce` with some default options.

I've had no luck trying to fix this issue myself.

**Comment by Dignifiedquire** [comment]
Released in [v0.13.5](https://github.com/karma-runner/karma/releases/tag/v0.13.5)

**Comment by Merott** [comment]
Although this fixes the issue for short delays, If you use a longer delay (e.g. 1000), you'll get a `TypeError: Cannot read property 'then' of undefined` error at [`common.js#12`](https://github.com/karma-runner/karma/blob/2c9be5a03f8de1311a62014966d5d87c94067ec0/lib/middleware/common.js#L12). It happens because the code in `common.js` gets hit before the promise is initialised inside of [`web-server.js#L37`](https://github.com/karma-runner/karma/blob/2c9be5a03f8de1311a62014966d5d87c94067ec0/lib/web-server.js#37).

That was why I suggested [throttling the event *after* its first emission](https://github.com/Merott/karma/commit/29835e0da51b70f59502b685d68e7054b41144e5).

**Comment by Dignifiedquire** [comment]
Thanks @Merott for pointing this out, this should be fixed in Should be fixed in [v0.13.6](https://github.com/karma-runner/karma/releases/tag/v0.13.6)
